File: 53-maximum-subarray/maximum-subarray.c
int maxSubArray(int* nums, int numsSize) {
    int max_sum = INT_MIN;
    int current_sum = 0;

    for (int i = 0; i < numsSize; i++) {
        current_sum += nums[i];
        if (current_sum > max_sum) {
            max_sum = current_sum;
        }
        if (current_sum < 0) {
            current_sum = 0;
        }
    }

    return max_sum;
}

File: 53-maximum-subarray/maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        current=nums[0]
        maxsum=nums[0]
        start,end=0,0
        temp=0
        for i in range(1,len(nums)):
            if nums[i]>current+nums[i]:
                current=nums[i]
                temp=i
            else:
                current+=nums[i]
            if current>maxsum:
                maxsum=current
                start=i
                end=i
        return maxsum
